Sources_v3/Controllers/CurrentUserController/CurrentUserController.swift

```diff
@@ -316,6 +316,85 @@ public extension _CurrentChatUserController {
             }
         }
     }
+    
+    /// Registers a device to the current user.
+    /// `setUser` must be called before calling this.
+    /// - Parameters:
+    ///   - token: Device token, obtained via `didRegisterForRemoteNotificationsWithDeviceToken` function in `AppDelegate`.
+    ///   - completion: Called when device is successfully registered, or with error.
+    func addDevice(token: Data, completion: ((Error?) -> Void)? = nil) {
+        guard let currentUserId = currentUser?.id else {
+            completion?(ClientError.CurrentUserDoesNotExist())
+            return
+        }
+        let deviceId = token.deviceToken
+        client
```

Comment:
nit: how about having `CurrentUserUpdater`?

Reply:
I think it makes sense to have such an entity, but considering the size of this PR I'd prefer doing it in another PR